src/Bicep.LangServer/Completions/ModuleReferenceCompletionProvider.cs

```diff
@@ -148,51 +147,57 @@ private bool IsOciModuleRegistryReference(string replacementText)
             return false;
         }
 
-        // Handles version completions for Microsoft Container Registries (MCR).
-        // I.e. completions starting with "br/" or "br:"
+        // Handles version completions for Microsoft Container Registries (MCR):
+        //
+        //   br/module/name:<CURSOR>
+        //   br:mcr.microsoft/bicep/module/name:<CURSOR>
+        //
+        // etc
         private async Task<IEnumerable<CompletionItem>> GetMCRModuleRegistryVersionCompletions(BicepCompletionContext context, string replacementText, Uri sourceFileUri)
         {
             if (!IsOciModuleRegistryReference(replacementText))
             {
                 return Enumerable.Empty<CompletionItem>();
             }
 
-            string? filePath;
+            string? modulePath;
 
             if (MCRModuleRegistryWithAlias.IsMatch(replacementText))
             {
                 var matches = MCRModuleRegistryWithAlias.Matches(replacementText);
-                filePath = matches[0].Groups["filePath"].Value;
+                modulePath = matches[0].Groups["filePath"].Value;
             }
             else if (MCRModuleRegistryWithoutAlias.IsMatch(replacementText))
             {
                 var matches = MCRModuleRegistryWithoutAlias.Matches(replacementText);
-                filePath = matches[0].Groups["filePath"].Value;
+                modulePath = matches[0].Groups["filePath"].Value;
             }
             else
             {
-                filePath = GetNonPublicMCRFilePathForVersionCompletion(replacementText, sourceFileUri);
+                modulePath = GetNonPublicMCRFilePathForVersionCompletion(replacementText, sourceFileUri);
             }
 
-            if (filePath is null)
+            if (modulePath is null)
             {
                 return Enumerable.Empty<CompletionItem>();
             }
 
             List<CompletionItem> completions = new List<CompletionItem>();
             replacementText = replacementText.TrimEnd('\'');
 
-            var versions = await publicRegistryModuleMetadataProvider.GetVersions(filePath);
-            for (int i = versions.Count() - 1; i >= 0; i--)
+            var versionInfos = await publicRegistryModuleMetadataProvider.GetVersions(modulePath);
+            for (int i = versionInfos.Count() - 1; i >= 0; i--)
```

Comment:
This is not introduced by this PR, but we should convert `versionInfos` to a list and replace `ElementAt` with index access to avoid iterating over the enumerable multiple times.

Reply:
a2fa2eb8220d7560f7b8ffd0adc62cde38dd379a